### PR TITLE
label: Invalidate empty or invalid value labels

### PR DESCRIPTION
**Pull request by 0xmichalis**

### CODE REVIEW

**Comment by bgrant0607** [comment]
ok to test

File: pkg/kubectl/cmd/label.go [hunk 1]
```diff
@@ -103,7 +104,7 @@ func parseLabels(spec []string) (map[string]string, []string, error) {
 	for _, labelSpec := range spec {
 		if strings.Index(labelSpec, "=") != -1 {
 			parts := strings.Split(labelSpec, "=")
-			if len(parts) != 2 {
+			if len(parts) != 2 || len(parts[1]) == 0 || !util.IsValidLabelValue(parts[1]) {
 				return nil, nil, fmt.Errorf("invalid label spec: %v", labelSpec)
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/label.go]: I know it's preexisting, but invalid user input should return [`UsageError`](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/util/helpers.go#L163). Do you mind fixing references in this file?

**Comment by jlowdermilk** [comment]
Users shouldn't be expected to remember the rules for a valid label spec. Let's include the [valid label regexp](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/validation.go#L54) in the help string.

File: pkg/kubectl/cmd/label.go (hunk 1 above)
Comment (kargakis) [inline pkg/kubectl/cmd/label.go]: Added this in the main RunLabel function.

**Comment by kargakis** [comment]
> Users shouldn't be expected to remember the rules for a valid label spec. Let's include the valid label regexp in the help string.

I've updated the help string with a simple explanation of what's needed for a valid label value.

File: pkg/kubectl/cmd/label.go [hunk 2]
```diff
@@ -25,12 +25,14 @@ import (
...
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
 	"github.com/spf13/cobra"
 )
 
 const (
 	label_long = `Update the labels on a resource.
 
+A valid label value is consisted of letters and/or numbers with a max length of %[1]d characters.
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/label.go]: Labels can also contain `-`, `.`, and`_`, which is why I was suggesting putting the regex into the help string. Rule is [here](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/util/validation.go#L27).

Comment (kargakis) [inline pkg/kubectl/cmd/label.go]: How about mentioning those characters as well?
```
A valid label value is consisted of letters, numbers, and some special characters (hyphen, dot, underscore) with a max length of 63 characters.
```

Imho it feels much more user-friendly than eg.
```
A valid label value has to comply with the ([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9] regural expression with a max length of 63 characters.
```

Comment (jlowdermilk) [inline pkg/kubectl/cmd/label.go]: I'm probably weird in that I prefer the regex. Fine with me to stick to English, so long as the rule is concisely and accurately stated. Label must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores, up to 63 characters.

Comment (kargakis) [inline pkg/kubectl/cmd/label.go]: updated

**Comment by jlowdermilk** [comment]
LGTM, thanks!

**Comment by thockin** [comment]
needs rebase

**Comment by kargakis** [comment]
rebased

**Pull request merged**